src/posts/posts-store.ts

```diff
@@ -0,0 +1,46 @@
+import { create } from "zustand";
+import { Post } from "./posts.types";
+
+type PostsStore = {
+  posts: Post[];
+  filteredTransactions: Post[];
+  statusFilter: string;
+  searchFilter: string;
+  setPosts: (posts: Post[]) => void;
+  setStatusFilter: (status: string) => void;
+  setSearchFilter: (search: string) => void;
+};
+
+export const usePostsStore = create<PostsStore>((set, get) => ({
+  posts: [],
+  filteredTransactions: [],
+  statusFilter: "",
+  searchFilter: "",
+  setPosts: (posts) => set({ posts, filteredTransactions: posts }),
+  setStatusFilter: (status) => {
+    const { posts, searchFilter } = get();
+    const searchLower = searchFilter.toLowerCase();
+    const filtered = posts.filter(
```

Comment:
always keep the Business logic outside the store, so move the search and filtering to its component, and just save the filtered result only